### PR TITLE
generate doc/requirements.txt

### PR DESCRIPTION
**Pull request by geekosaur**
As suggested in https://github.com/haskell/cabal/pull/10030#pullrequestreview-2064505535

**Template B: This PR does not modify behaviour or interface**

*E.g. the PR only touches documentation or tests, does refactorings, etc.*

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Is this a PR that fixes CI? If so, it will need to be backported to older cabal release branches (ask maintainers for directions).

### CODE REVIEW

**Review by ulysses4ever — COMMENTED** [review]
I'd expect some updates to documentation. e.g. `doc/README.md`. Maybe the Makefile.

**Comment by geekosaur** [comment]
Yes, this is still WIP, let me switch it to draft.

**Comment by geekosaur** [comment]
Actually, I'm not sure we want to update the Makefile; it's going to be _more_ necessary with this change, if someone wants to build docs locally. `doc/README.md` and possibly `CONTRIBUTING` would need to be updated.

**Comment by geekosaur** [comment]
Okay, the basics now work. Documentation is next (tomorrow, I should go to bed…).

**Comment by ffaf1** [comment]
This is magnificent, many thanks!

**Comment by geekosaur** [comment]
First cut at documentation added.

**Comment by geekosaur** [comment]
1. I changed the doc build rules to use the `Makefile` rule, so that is tested.
2. I do _not_ expect that this needs to be backported, but will leave the final decision to wiser heads.

**Review by ffaf1 — APPROVED** [review]

File: doc/README.md [hunk 1]
```diff
@@ -11,48 +11,34 @@ http://cabal.readthedocs.io/
 
 ### How to build it
 
-Building the documentation requires Python 3 and PIP. Run the following command either from the root of the cabal repository or from the `docs/` subdirectory:
+Building the documentation requires Python 3, PIP, and pip-tools. Run the following command either from the root of the cabal repository or from the `docs/` subdirectory:
```
Comment (ulysses4ever) [inline doc/README.md:14]:
```suggestion
Building the documentation requires Python 3, PIP, and `pip-tools` (see below how to get it). Run the following command either from the root of the cabal repository or from the `docs/` subdirectory:
```

File: doc/README.md [hunk 2]
```diff
@@ -11,48 +11,34 @@ http://cabal.readthedocs.io/
...
+> make users-guide-requirements
 ```
 
-One way to make sure the dependencies are reasonably up to date
-is to remove `requirements.txt` and regenerate it as described
-above. But in some cases you may have to add a bound manually
-to `requirements.in`, e.g. `requests >= 2.31.0`.
+either from the root of the cabal repository or from the `docs/` subdirectory. You will need to do this before building documentation the first time, but should only need to repeat it after a `git clean` or if the dependencies change.
+
+In some cases you may have to add a bound manually to `requirements.in`, e.g. `requests >= 2.31.0`.
```
Comment (ulysses4ever) [inline doc/README.md:41]:
```suggestion
In some cases, you may have to add a bound manually to `requirements.in`, e.g. `requests >= 2.31.0`.
```

**Review by ulysses4ever — APPROVED** [review]
Long awaited improvement, thank you very much!

**Comment by geekosaur** [comment]
The reason this needs to be backported is that backports of subsequent changes to `doc/requirements.in` won't update `doc/requirements/.txt` on 3.12, so someone would need to manually do so for the backport to actually do anything.

**Comment by ulysses4ever** [comment]
@mergify backport 3.12

**Pull request merged**